### DISCUSSION TITLE
The table I'm searching is fixed, and it may return values of multiple types. I want to use a generic to receive it, but the Rust compiler keeps giving me errors, and I don't know how to write the constraints for the generic.

### DISCUSSION THREAD

**Discussion by li-up-up-up** [diesel-rs/diesel#4164] [Q&A]
```rust
pub fn find_article_list_by_filter<T>(
    search: Box<PageReq<SearchArticle, ArticleOrder>>,
) -> anyhow::Result<PageHelper<T>>
where
    T:  XXXXX,
{
    let mut search_info = tb_article::table.select(T::as_select()).into_boxed();
```

**Comment by weiznich** [comment]
I generally advice people not to write this kind of generic code in applications as is requires complex bounds and a good understanding of rusts trait system and diesels trait setup. I do not provide help writing such code as you need to understand yourself what's going on there to be able to maintain the result in the long run.

**Reply by li-up-up-up** [reply]
However, this function is very large and has several types. Using generics is safer, and writing the same code multiple times is not meaningful.

**Reply by weiznich** [reply]
I did not write that it cannot be done nor did I forbid it to be done. I just wrote that I do not have the capacity to help you writing that code and maintaining it later. So again: You are on your own if you want to go down that way.
